src/network/core/address.cpp

```diff
@@ -301,7 +306,12 @@ static SOCKET ConnectLoopProc(addrinfo *runp)
 
 	if (!SetNoDelay(sock)) DEBUG(net, 1, "[%s] setting TCP_NODELAY failed", type);
 
-	if (connect(sock, runp->ai_addr, (int)runp->ai_addrlen) != 0) {
+	int err = connect(sock, runp->ai_addr, (int)runp->ai_addrlen);
+#ifndef __EMSCRIPTEN__
+	if (err != 0) {
+#else
+	if (err != 0 && errno != EINPROGRESS) {
```

Comment:
Presumably this is because Emscripten doesn't define `EINPROGRESS` ? Might be cleaner to define it to something (0 would work?) than the 2 separate if statements

Also add a comment explaining why it's necessary

Reply:
Emscripten doesn't support synchronous sockets, so they are always created nonblocking. (thus connect always returns EINPROGRESS, and that cannot be interpreted as error)

Reply:
Fair enough, just a comment explaining as such then